cmd/kubeadm/app/discovery/file/file.go

```diff
@@ -87,6 +87,10 @@ func ValidateConfigInfo(config *clientcmdapi.Config, clustername string, discove
 			"", // no user provided
 			currentCluster.CertificateAuthorityData,
 		)
+		currentClusterName = clustername
+		if len(currentCluster.CertificateAuthorityData) == 0 {
+			config.Clusters[clustername].InsecureSkipTLSVerify = true
```

Comment:
@neolit123 This is needed. Because we create a new `config` here, but `kubeconfigutil.CreateBasic`(line 84) only pick up `currentCluster.Server` and `currentCluster.CertificateAuthorityData` filed from user input kubeconfig. Even if the user is turned on `InsecureSkipTLSVerify` via discovery kubeconfig, `kubeconfigutil.CreateBasic` will always set it as false. This will cause problems when the discovery kubeconfig does not contain CA information.

I prefer to just remove the `kubeconfigutil.CreateBasic` logic to respect the user's input as much as possible. Then we can leave `InsecureSkipTLSVerify` as it is. WDYT?

Reply:
yes, perhaps we should modify createbasic to accept a user config pointer or nil in case it creates a basic config.